float32 a
float32 x_c
float32 y_c
float32 sigma_x
float32 sigma_y